source/isaaclab/isaaclab/scene/interactive_scene.py

```diff
@@ -479,38 +476,93 @@ def reset_to(
             root_velocity = asset_state["root_velocity"].clone()
             rigid_object.write_root_pose_to_sim(root_pose, env_ids=env_ids)
             rigid_object.write_root_velocity_to_sim(root_velocity, env_ids=env_ids)
+
+        # write data to simulation to make sure initial state is set
+        # this propagates the joint targets to the simulation
         self.write_data_to_sim()
 
-    def write_data_to_sim(self):
-        """Writes the data of the scene entities to the simulation."""
-        # -- assets
-        for articulation in self._articulations.values():
-            articulation.write_data_to_sim()
-        for deformable_object in self._deformable_objects.values():
-            deformable_object.write_data_to_sim()
-        for rigid_object in self._rigid_objects.values():
-            rigid_object.write_data_to_sim()
-        for rigid_object_collection in self._rigid_object_collections.values():
-            rigid_object_collection.write_data_to_sim()
+    def get_state(self, is_relative: bool = False) -> dict[str, dict[str, dict[str, torch.Tensor]]]:
+        """Returns the state of the scene entities.
 
-    def update(self, dt: float) -> None:
-        """Update the scene entities.
+        Based on the type of the entity, the state comprises of different components.
+
+        * For an articulation, the state comprises of the root pose, root velocity, and joint position and velocity.
+        * For a deformable object, the state comprises of the nodal position and velocity.
+        * For a rigid object, the state comprises of the root pose and root velocity.
+
+        The returned state is a dictionary with the following format:
+
+        .. code-block:: python
+
+            {
```

Comment:
@kellyguo11 Is there a reason why there is a separation for articulation/rigid bodies and deformable obejcts? The scene entitty could have been refered to directly without this additional mesting at the top-most level. IMO we should get rid of it as it doesn't add any value.